src/utils/responses.ts

```diff
@@ -202,6 +202,8 @@ export const StatusCodes = {
 export const StatusResponseSchema = z.object({
 	message: z.string()
 		.describe('A message reporting the status of the operation.'),
+	warning: z.string().optional()
+		.describe('A warning message regarding the status of the operation.'),
```

Comment:
```suggestion
	warning: z.array(z.record(z.string(), z.string()))
		.optional()
		.describe('A warning message regarding the status of the operation.'),
```

I think here we can do the same as `errors` and provide structured data (i.e.: an object), that way it is easier to parse and handle specific values instead of delaing with string parsing.